Split DefaultImageScalingFactory into more methods for flexibility

mauritsvanrees opened the pull request:
plone.app.tiles has its [own image scale factory](https://github.com/plone/plone.app.tiles/blob/3.2.2/plone/app/tiles/imagescaling.py#L110).
It [passes this around](https://github.com/plone/plone.app.tiles/blob/3.2.2/plone/app/tiles/imagescaling.py#L180) in a [deprecated way](https://github.com/plone/plone.scale/blob/3.1.2/plone/scale/storage.py#L207-L212).
Its scale factory does almost the same as the DefaultImageScalingFactory, although it is structured differently, as a function instead of a class.
Result is that the tiles scale factory is missing some features and fixes from the default, at least support for SVG and passing a quality parameter.

The only *intended* difference of the tile scale factory, is that it gets the original field value from the [tile data](https://github.com/plone/plone.app.tiles/blob/3.2.2/plone/app/tiles/imagescaling.py#L114),
instead of a [field on the context](https://github.com/plone/plone.namedfile/blob/5.5.1/plone/namedfile/scaling.py#L214).

The current commit makes the DefaultImageScalingFactory work for tiles.
Next step would be to update plone.app.tiles to use this.  This mostly involves removing code.

(Maybe something similar would be useful/needed for images in portlets, but I did not check. Do scales of images fields in portlets even work today?)

This will also make future support for focal points easier: there will be just one scaling factory to fix, instead of two.

For me it would be fine to include this in Plone 5.2: I am using this code in a client project on 5.2, and it is working fine. But I am biased. Also fine to do this in 6.0 only.

mister-roboto commented:
@mauritsvanrees thanks for creating this Pull Request and help improve Plone!

To ensure that these changes do not break other parts of Plone, the Plone test suite matrix needs to pass.

Whenever you feel that the pull request is ready to be tested, either start all jenkins jobs pull requests by yourself, or simply add a comment in this pull request stating:

```
@jenkins-plone-org please run jobs
```

With this simple comment all the jobs will be started automatically.

Happy hacking!

mauritsvanrees commented:
@jenkins-plone-org please run jobs

jensens commented:
Well, we never use binary data in tiles here, but always store images and files in types and reference them. So I never cared. 

Anyway, I do not like this PR. plone.tiles is an addon, well, even if its Plone "ecosystem". I do not think we should blow up core code for addons. We can make the core module more flexible and have the specific implementation then using a hook or an adapter. 

So -1 from me for this one.

mauritsvanrees commented:
>  I do not think we should blow up core code for addons. We can make the core module more flexible and have the specific implementation then using a hook or an adapter.

Fair enough. I have updated the PR to not have tile-specific code. Instead I split the `__call__` method up into several methods, making it smaller, and easier to override only parts of it. Is this better?

@jenkins-plone-org please run jobs
